src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -2105,9 +2104,29 @@ namespace QgsWms
     return mWmsParameters.value( QgsWmsParameter::WITH_GEOMETRY ).toBool();
   }
 
+  QString QgsWmsParameters::withMapTipAsString() const
+  {
+    return mWmsParameters.value( QgsWmsParameter::WITH_MAPTIP ).toString();
+  }
+
   bool QgsWmsParameters::withMapTip() const
   {
-    return mWmsParameters.value( QgsWmsParameter::WITH_MAPTIP ).toBool();
+    const QString mStr = withMapTipAsString();
+
+    if ( mStr.startsWith( QLatin1String( "true" ), Qt::CaseInsensitive ) )
```

Comment:
I think you should also check for:
ON
YES
1

Reply:
done in https://github.com/qgis/QGIS/pull/55823/commits/e13d9aab32affa2eb6f95fe7b137ed995613bbe6